docs/how-to/create-epics-ioc-with-ca-and-pva.md

```diff
@@ -0,0 +1,951 @@
+- Verify device IP address and port are correct
+- Check network connectivity: `ping <device-ip>`
+- Ensure device is powered on and responding
+- Check device documentation for correct protocol
+
+### Slow Updates
+
+**Problem**: PVs update slower than expected
+
+**Solutions**:
+- Check `update_period` in `io_ref` (lower = faster updates)
+- Reduce number of attributes with fast updates
+- Optimize device communication (batch queries if possible)
+- Consider if device can handle higher query rate
+
+### Import Errors
+
+**Problem**: `ModuleNotFoundError: No module named 'p4p'` or similar
+
+**Solutions**:
+```bash
+# Reinstall with extras
+pip install --force-reinstall 'fastcs[ca,pva]'
+
+# Or install individually
+pip install pythonSoftIOC
+pip install p4p
+```
+
+### GUI Not Generating
+
+**Problem**: `.bob` file not created
+
+**Solutions**:
+- Check `output_path` is valid and writable
+- Ensure directory exists (or will be created)
+- Look for errors in IOC startup logs
+- Verify `gui` parameter is passed to transport
+
+## Best Practices
+
+### PV Naming Conventions
+
+```python
+# Good: Clear, hierarchical names
+MY-DEVICE:Temperature
+MY-DEVICE:PowerSupply:Voltage
+MY-DEVICE:Motor1:Position
+
+# Avoid: Abbreviations, unclear names
+MY-DEV:TEMP
+MY-DEVICE:PSU1_V
+MY-DEVICE:M1POS
+```
+
+### When to Use CA vs PVA
+
+**Use both** when:
+- Supporting legacy and modern clients
+- During transition periods
+- Maximum compatibility needed
+
+**Use CA only** when:
+- Only legacy clients exist
+- Proven stability required
+- Simple scalar values only
+
+**Use PVA only** when:
+- Only modern clients (Phoebus)
+- Need structured data (Tables)
+- Better performance required
+- New deployment without legacy
+
+### Performance Considerations
+
+```python
+# Good: Different update rates for different needs
+class MyController(Controller):
+    critical_temp: AttrR = AttrR(
+        Float(),
+        io_ref=IORef(command="TEMP", update_period=0.1)  # Fast: 10 Hz
+    )
+
+    device_id: AttrR = AttrR(
+        String(),
+        io_ref=IORef(command="ID", update_period=3600.0)  # Slow: 1/hour
+    )
+```
+
+### Error Handling Patterns
+
+```python
+class RobustAttributeIO(AttributeIO):
+    async def update(self, attr):
+        try:
+            response = await self._connection.send_query(
+                f"{attr.io_ref.command}?\r\n",
+                timeout=2.0
+            )
+            value = attr.dtype(response.strip())
+            await attr.update(value)
+        except asyncio.TimeoutError:
+            # Log but don't crash - old value remains
+            logger.warning(f"Timeout reading {attr.io_ref.command}")
+        except ValueError as e:
+            # Bad response format
+            logger.error(f"Invalid response: {e}")
+            raise  # Re-raise to stop updates
+        except ConnectionError:
+            # Connection lost - will try to reconnect
+            logger.error("Device connection lost")
+            raise
+```
```

Comment:
It seems to have figured out that raising in an update loop stops it, but then asserts that the connection error arm will reconnect, which it will not.

Overall this doesn't make a huge amount of sense.